diff --git a/CMakeLists.txt b/CMakeLists.txt
index c749abb..051960d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,10 @@ ENDIF()
 
 #add_subdirectory (src)
 
+IF("${CMAKE_TOOLCHAIN_FILE}" MATCHES "android.toolchain.cmake$")
+    include(${CMAKE_TOOLCHAIN_FILE})
+ENDIF()
+
 SET (LIB_TYPE STATIC)
 IF (BUILD_SHARED_LIBS)
   SET (LIB_TYPE SHARED)
@@ -59,8 +63,6 @@ TARGET_LINK_LIBRARIES(${LIB_NAME} m ${PTHREADS_LIBRARY}
 )
 
 SET_TARGET_PROPERTIES(${LIB_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION} SOVERSION ${${PROJECT_NAME}_MAJOR_VERSION})
-install(TARGETS ${LIB_NAME} LIBRARY DESTINATION ${LIB_DESTINATION})
-install(FILES ${H_FILES} DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Devel)
 
 ENABLE_TESTING()
 
diff --git a/fitscore.c b/fitscore.c
index 4c90df0..5e59cc1 100644
--- a/fitscore.c
+++ b/fitscore.c
@@ -9230,7 +9230,7 @@ int ffc2rr(const char *cval,   /* I - string representation of the value */
 
     if (!decimalpt) { /* only do this once for efficiency */
        lcc = localeconv();   /* set structure containing local decimal point symbol */
-       decimalpt = *(lcc->decimal_point);
+//       decimalpt = *(lcc->decimal_point);
     }
 
     errno = 0;
@@ -9300,7 +9300,7 @@ int ffc2dd(const char *cval,   /* I - string representation of the value */
 
     if (!decimalpt) { /* only do this once for efficiency */
        lcc = localeconv();   /* set structure containing local decimal point symbol */
-       decimalpt = *(lcc->decimal_point);
+//       decimalpt = *(lcc->decimal_point);
     }
    
     errno = 0;
