diff --git a/CMakeLists.txt b/CMakeLists.txt
index 47f3869..f5df332 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,9 +114,16 @@ INCLUDE(MacroLogFeature)
 INCLUDE(MacroOptionalFindPackage)
 INCLUDE(MacroJPEG)
 
+IF(${CMAKE_TOOLCHAIN_FILE} MATCHES ".*toolchain-android.cmake")
+    include(${CMAKE_TOOLCHAIN_FILE})
+ENDIF()
+INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/android-hack)
+
 # Math library check
 
-IF(NOT WIN32)
+IF(${CMAKE_TOOLCHAIN_FILE} MATCHES ".*toolchain-android.cmake")
+    set(MATH_LIBRARY -lm) # toolchain-android.cmake can't find libm.so
+ELSEIF(NOT WIN32)
     FIND_LIBRARY(MATH_LIBRARY m)
 ENDIF()
 
diff --git a/android-hack/swab.h b/android-hack/swab.h
new file mode 100644
index 0000000..061c61e
--- /dev/null
+++ b/android-hack/swab.h
@@ -0,0 +1,6 @@
+
+#pragma once
+
+#include <stdint.h>
+
+void swab(const void *from, void*to, ssize_t n);
diff --git a/internal/dcraw_common.cpp b/internal/dcraw_common.cpp
index 608641a..0f991e4 100644
--- a/internal/dcraw_common.cpp
+++ b/internal/dcraw_common.cpp
@@ -19,6 +19,24 @@ it under the terms of the one of two licenses as you choose:
 */
 
 
+#ifdef ANDROID
+#include <swab.h>
+
+#include <stdint.h>
+#include <asm/byteorder.h>
+
+void swab(const void *from, void*to, ssize_t n)
+{
+  ssize_t i;
+
+  if (n < 0)
+    return;
+
+  for (i = 0; i < (n/2)*2; i += 2)
+    *((uint16_t*)to+i) = __arch__swab16(*((uint16_t*)from+i));
+}
+#endif
+
 #include <math.h>
 #define CLASS LibRaw::
 #include "libraw/libraw_types.h"
diff --git a/src/libraw_cxx.cpp b/src/libraw_cxx.cpp
index c977302..2112ba8 100644
--- a/src/libraw_cxx.cpp
+++ b/src/libraw_cxx.cpp
@@ -16,6 +16,10 @@ it under the terms of the one of two licenses as you choose:
 
  */
 
+#if ANDROID
+#include <swab.h>
+#endif
+
 #include <math.h>
 #include <errno.h>
 #include <float.h>
